### ISSUE TITLE
aws - sts regional endpoints - some services only support global regions

### ISSUE THREAD

**Issue by PratMis** [cloud-custodian/cloud-custodian#5023]
['/usr/local/bin/custodian', 'run', '--region', 'us-east-1', '-c', 'policy.yml', '--assume=arn:aws:iam::xxx', '-m', 'aws://?namespace=Default' '-s', 'output', '--dryrun', '-v']
Traceback:
```
2019-10-31 22:12:24,498: custodian.commands:DEBUG Loaded file policy.yml
Traceback (most recent call last):
File "/usr/local/bin/custodian", line 8, in <module>
sys.exit(main())
File "/usr/local/lib/python3.7/site-packages/c7n/cli.py", line 374, in main
command(config)
File "/usr/local/lib/python3.7/site-packages/c7n/commands.py", line 141, in _load_policies
return f(options, list(policies))
File "/usr/local/lib/python3.7/site-packages/c7n/commands.py", line 275, in run
local_session(clouds['aws']().get_session_factory(options))
File "/usr/local/lib/python3.7/site-packages/c7n/utils.py", line 274, in local_session
s = factory()
File "/usr/local/lib/python3.7/site-packages/c7n/credentials.py", line 57, in __call__
region or self.region, self.external_id)
File "/usr/local/lib/python3.7/site-packages/c7n/credentials.py", line 115, in assumed_session
metadata=refresh(),
File "/usr/local/lib/python3.7/site-packages/c7n/credentials.py", line 106, in refresh
session, region).assume_role, **parameters)['Credentials']
File "/usr/local/lib/python3.7/site-packages/c7n/utils.py", line 390, in _retry
return func(*args, **kw)
File "/usr/local/lib/python3.7/site-packages/botocore/client.py", line 357, in _api_call
return self._make_api_call(operation_name, kwargs)
File "/usr/local/lib/python3.7/site-packages/botocore/client.py", line 648, in _make_api_call
operation_model, request_dict, request_context)
File "/usr/local/lib/python3.7/site-packages/botocore/client.py", line 667, in _make_request
return self._endpoint.make_request(operation_model, request_dict)
File "/usr/local/lib/python3.7/site-packages/botocore/endpoint.py", line 102, in make_request
return self._send_request(request_dict, operation_model)
File "/usr/local/lib/python3.7/site-packages/botocore/endpoint.py", line 132, in _send_request
request = self.create_request(request_dict, operation_model)
File "/usr/local/lib/python3.7/site-packages/botocore/endpoint.py", line 116, in create_request
operation_name=operation_model.name)
File "/usr/local/lib/python3.7/site-packages/botocore/hooks.py", line 356, in emit
return self._emitter.emit(aliased_event_name, **kwargs)
File "/usr/local/lib/python3.7/site-packages/botocore/hooks.py", line 228, in emit
return self._emit(event_name, kwargs)
File "/usr/local/lib/python3.7/site-packages/botocore/hooks.py", line 211, in _emit
response = handler(**kwargs)
File "/usr/local/lib/python3.7/site-packages/botocore/signers.py", line 90, in handler
return self.sign(operation_name, request)
File "/usr/local/lib/python3.7/site-packages/botocore/signers.py", line 149, in sign
auth = self.get_auth_instance(**kwargs)
File "/usr/local/lib/python3.7/site-packages/botocore/signers.py", line 233, in get_auth_instance
raise botocore.exceptions.NoRegionError()
botocore.exceptions.NoRegionError: You must specify a region.
end of run
```

**Comment by kapilt** [comment]
what are the docker run parameters ?

**Comment by kapilt** [comment]
works fine afaics

```
$ docker run -e AWS_ACCESS_KEY_ID -e AWS_SECRET_ACCESS_KEY   -v $PWD/rds.yml:/rds.yml cloudcustodian/c7n:latest run --region us-east-1 -c /rds.yml -m aws://?namespace=Default -s output --dryrun -v
2019-11-03 12:07:33,265: custodian.commands:DEBUG Loaded file /rds.yml. Contains 1 policies
2019-11-03 12:07:33,611: custodian.output:DEBUG Storing output with <LogFile file://output/cluster-tags/custodian-run.log>
2019-11-03 12:07:33,631: custodian.policy:DEBUG Running policy:cluster-tags resource:rds-cluster region:us-east-1 c7n:0.8.45.1
2019-11-03 12:07:38,170: custodian.cache:INFO Generating Cache directory: /home/custodian/.cache.
2019-11-03 12:07:38,172: custodian.resources.rdscluster:DEBUG Filtered from 2 to 2 rdscluster
2019-11-03 12:07:38,172: custodian.policy:INFO policy:cluster-tags resource:rds-cluster region:us-east-1 count:2 time:4.54
2019-11-03 12:07:38,175: custodian.policy:DEBUG dryrun: skipping actions
```

**Comment by PratMis** [comment]
Investigating more! I'll report back as soon as I find something. Thanks

**Comment by PratMis** [comment]
I suspect that its happening due to #4987 . Passing AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY  as docker env variables wouldn't let you hit get_sts_client function. Was able to reproduce the issue with --assume=arn:aws:iam::xxxxxx. 
It also complains about some of the global resources. Like: `c7n.policies:WARNING policy:rpolicy_name resources:r53domain not available in region:us-west-2`

**Comment by kapilt** [comment]
thanks for debugging, okay we're going to need keep a set of services/regions that don't support endpoints.

**Comment by kapilt** [comment]
I think we should switch the default on this feature to off

**Comment by PratMis** [comment]
Thanks! Was fixed via #5043 and #5052
